ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/Smoke_Test.java

```diff
@@ -1136,4 +1136,78 @@ public SeekableByteChannel buildChannel(final String key) throws IOException {
             deleteAllContents(client, bucketName);
         }
     }
+
+    @Test
+    public void TestPlusCharacterInQueryParam() throws IOException, SignatureException, XmlProcessingException {
+        final String bucketName = "TestPlusCharacterInQueryParam";
+        final String objectName = "Test+Plus+Character";
+        try {
+            final Ds3ClientHelpers helpers = Ds3ClientHelpers.wrap(client);
+
+            helpers.ensureBucketExists(bucketName);
+
+            final List<Ds3Object> objs = Lists.newArrayList(new Ds3Object(objectName, 10));
+
+            final Ds3ClientHelpers.Job job = helpers.startWriteJob(bucketName, objs);
+
+            job.transfer(new Ds3ClientHelpers.ObjectChannelBuilder() {
+                @Override
+                public SeekableByteChannel buildChannel(final String key) throws IOException {
+
+                    final byte[] randomData = IOUtils.toByteArray(new RandomDataInputStream(124345, 10));
+                    final ByteBuffer randomBuffer = ByteBuffer.wrap(randomData);
+
+                    final ByteArraySeekableByteChannel channel = new ByteArraySeekableByteChannel(10);
+                    channel.write(randomBuffer);
+
+                    return channel;
+                }
+            });
+
+            final GetObjectsSpectraS3Response getObjectsSpectraS3Response = client
+                    .getObjectsSpectraS3(new GetObjectsSpectraS3Request().withName(objectName));
+
+            assertThat(getObjectsSpectraS3Response.getS3ObjectListResult().getS3Objects().size(), is(1));
```

Comment:
Same comment as with the other pull requests,  please verify that the object you get back is the correct one.